src/System.Diagnostics.TextWriterTraceListener/tests/XmlWriterTraceListenerTests.cs

```diff
@@ -0,0 +1,501 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.CodeDom;
+using System.CodeDom.Compiler;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using Xunit;
+
+namespace System.Diagnostics.TextWriterTraceListenerTests
+{
+    public class XmlWriterTraceListenerTests : FileCleanupTestBase
+    {
+        private readonly string _processName;
+
+        public XmlWriterTraceListenerTests()
+        {
+            using (var process = Process.GetCurrentProcess())
+            {
+                _processName = process.ProcessName;
+            }
+        }
+
+        public static IEnumerable<object[]> ConstructorsTestData()
+        {
+            yield return new object[] { new XmlWriterTraceListener(new MemoryStream()), string.Empty };
+            yield return new object[] { new XmlWriterTraceListener(new StreamWriter(new MemoryStream())), string.Empty };
+            yield return new object[] { new XmlWriterTraceListener(Path.GetTempFileName()), string.Empty };
+            yield return new object[] { new XmlWriterTraceListener(new MemoryStream(), "MemoryStreamListener"), "MemoryStreamListener" };
+            yield return new object[] { new XmlWriterTraceListener(new StreamWriter(new MemoryStream()), "StreamWriterListener"), "StreamWriterListener" };
+            yield return new object[] { new XmlWriterTraceListener(Path.GetTempFileName(), "FileNameListener"), "FileNameListener" };
+        }
+
+        [Theory]
+        [MemberData(nameof(ConstructorsTestData))]
+        public void SingleArgumentConstructorTest(XmlWriterTraceListener listener, string expectedName)
+        {
+            Assert.Equal(expectedName, listener.Name);
+        }
+
+        [Fact]
+        public void ConstructorThrows_ArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() => new XmlWriterTraceListener((Stream)null));
+            Assert.Throws<ArgumentNullException>(() => new XmlWriterTraceListener((TextWriter)null));
```

Comment:
Any parameter names that could/should be validated as well?  Same goes for other such asserts.